apps/wpcom-block-editor/src/wpcom/features/tracking/wpcom-block-editor-list-view-select.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * External dependencies
+ */
+import { select } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import tracksRecordEvent from './track-record-event';
+
+/**
+ * Return the event definition object to track `wpcom_block_editor_list_view_select`.
+ *
+ * @returns {import('./types').DelegateEventHandler} event object definition.
+ */
+export default () => ( {
+	id: 'wpcom-block-editor-list-view-select',
+	selector: '[aria-label="Block navigation structure"] [role="row"]:not(.is-selected) button',
+	type: 'click',
+	capture: true,
+	handler: async () => {
+		const { getSelectedBlock } = select( 'core/block-editor' );
+
+		const blockBefore = getSelectedBlock();
+		// Wait until the selected block changes.
+		while ( blockBefore?.clientId === getSelectedBlock()?.clientId ) {
+			await new Promise( ( resolve ) => setTimeout( resolve, 10 ) );
+		}
+
+		const currentBlock = getSelectedBlock();
```

Comment:
Does running this on bubble instead of capture allow us to avoid awaiting the block change?

Reply:
It doesn't work with bubbling at all. I think the event handler in Gutenberg prevents it.